src/app/search/search-config.test.ts

```diff
@@ -16,23 +17,143 @@ describe('buildSearchConfig', () => {
+  const queryCases: [filterString: string, valid: boolean][] = [
+    ['tag:favorite', true],
+    ['tag:none', true],
+    ['tag:any', false],
+    ['is:tag', false],
+    ['tag:<5', false],
+    ['tag:recovery:17', false],
+  ];
+
+  test.each(queryCases)('query filter %s - validity %s', (filterString, valid) =>
+    expect(parseAndValidateQuery(filterString, searchConfig).valid).toBe(valid)
+  );
+
+  const freeformCases: [filterString: string, valid: boolean][] = [
+    ['notes:#hashtag', true],
+    ['notes:verbatim:colon', true],
+    ['notes"with spaces"', true],
+    ['notes:<5', true],
+    ['is:notes', false],
+  ];
+
+  test.each(freeformCases)('freeform filter %s - validity %s', (filterString, valid) =>
+    expect(parseAndValidateQuery(filterString, searchConfig).valid).toBe(valid)
+  );
+
+  // `masterwork` is a complicated filter with three different formats
+  const mixedCases: [filterString: string, valid: boolean][] = [
+    ['is:masterwork', true],
+    ['masterwork:range', true],
+    ['masterwork:<5', true],
+    ['masterwork:=5', true],
+    ['masterwork:5', true],
+    ['masterwork:rnage', false],
+    ['masterwork:<range', false],
+    ['masterwork:range:5', false],
+  ];
+
+  test.each(mixedCases)('mixed filter %s - validity %s', (filterString, valid) =>
+    expect(parseAndValidateQuery(filterString, searchConfig).valid).toBe(valid)
+  );
+
+  const statCases: [filterString: string, valid: boolean][] = [
+    ['stat:recovery:5', true],
+    ['stat:recovery:<=5', true],
+    ['stat:recovery+discipline:>=5', true],
+    ['stat:highest&secondhighest:>20', true],
+    ['stat:highest&recovery+strength&strength&range:>20', true],
+    ['basestat:range:>50', true],
+
+    ['stat:badstat:>20', false],
+    ['stat:badstat&badcombo:>20', false],
+    ['stat:too&+many:>20', false],
+    ['is:stat', false],
+    ['stat:recovery', false],
+    ['stat:=5', false],
+  ];
+
+  test.each(statCases)('stat filter %s - validity %s', (filterString, valid) =>
+    expect(parseAndValidateQuery(filterString, searchConfig).valid).toBe(valid)
+  );
+
+  const rangeCases: [filterString: string, valid: boolean][] = [
+    ['count:2', true],
+    ['count:=2', true],
+    ['count:<=2', true],
+    ['count:<=2.5', true],
+
+    ['is:count', false],
+    ['count:count', false],
+    ['count:recovery', false],
+    ['count:<=2:=2', false],
+    ['count:<2>', false],
+  ];
+
+  test.each(rangeCases)('search string %s - validity %s', (filterString, valid) =>
+    expect(parseAndValidateQuery(filterString, searchConfig).valid).toBe(valid)
+  );
+
+  const overloadRangeCases: [filterString: string, valid: boolean][] = [
+    ['season:worthy', true],
+    ['season:<=worthy', true],
+    ['season:=worthy', true],
+    ['season:>=arrival', true],
+    ['season:222', true],
+    ['season:=10', true],
+    ['season:>2.5', true],
+
+    ['is:season', false],
+    // DIM used to parse this as `season:11` because `redwar` is `1`...
+    ['season:1redwar', false],
```

Comment:
That this worked is beautiful in a way but I think it's acceptable to deny this now.